web/src/lib/components/resume/Editor.svelte

```diff
@@ -0,0 +1,480 @@
+    let resumeCreatedAt = 0;
+
+    let newSkill = new UserResumeSkill();
+    let newPortfolio = new UserResumePortfolio();
+    let newEducation: UserResumeEducation | null = null;
+    let newExperience: UserResumeExperience | null = null;
+    let newAchievement: UserResumeAchievement | null = null;
+
+    let availableForWork = false;
+
+    $: saveButtonActive = $user && !saving;
+
+    function addSkill() {
+        if (newSkill.validate()) {
+            resume.skills = [...resume.skills, newSkill];
+            newSkill = new UserResumeSkill();
+        }
+    }
+
+    function removeSkill(i) {
+        resume.skills = resume.skills.slice(0, i).concat(resume.skills.slice(i + 1));
+    }
+
+    function addPortfolio() {
+        if (newPortfolio.validate()) {
+            resume.portfolio = [...resume.portfolio, newPortfolio];
+            newPortfolio = new UserResumePortfolio();
+        }
+    }
+
+    function removePortfolio(i) {
+        resume.portfolio = resume.portfolio.slice(0, i).concat(resume.portfolio.slice(i + 1));
+    }
+
+    function addEducation() {
+        if (newEducation && newEducation.validate()) {
+            resume.education = [...resume.education, newEducation];
+            newEducation = new UserResumeEducation();
+        }
+    }
+
+    function removeEducation(i) {
+        resume.education = resume.education.slice(0, i).concat(resume.education.slice(i + 1));
+    }
+
+    function addExperience() {
+        if (newExperience && newExperience.validate()) {
+            resume.experience = [...resume.experience, newExperience];
+            newExperience = new UserResumeExperience();
+        }
+    }
+
+    function removeExperience(i) {
+        resume.experience = resume.experience.slice(0, i).concat(resume.experience.slice(i + 1));
+    }
+
+    function addAchievement() {
+        if (newAchievement && newAchievement.validate()) {
+            resume.achievements = [...resume.achievements, newAchievement];
+            newAchievement = new UserResumeAchievement();
+        }
+    }
+
+    function removeAchievement(i) {
+        resume.achievements = resume.achievements.slice(0, i).concat(resume.achievements.slice(i + 1));
+    }
+
+    let saving = false;
+    async function save() {
+        if ($user) {
+            saving = true;
+
+            let notified = false;
+
+            await publishResume(pool, $user, resume,
+                () => {
+                    if (!notified) {
+                        notified = true;
+                        saving = false;
+                        Info.set("Your résumé has been saved!");
+                        onEditFinished();
+                    }
+                });
+            // TODO: callback when no relay accepted our resume (so successCB was never called)
+        }
+    }
+
+    onMount(async () => {
+        if ($user) {
+            subscribeResume(pool, await getPreferredPublicKey($user.nostr_private_key),
```

Comment:
So if we're supporting this feature only when you have a Nostr extension, and we're not using our API in any way, maybe this can be simplified, and we don't need that the user is logged in?